### PR TITLE
Feat/uploader iteration

### PR DESCRIPTION
**Pull request by ferransimon**

### CODE REVIEW

File: components/atom/upload/src/index.js [hunk 1]
```diff
@@ -105,7 +133,21 @@ AtomUpload.propTypes = {
...
+   * Allow drag 'n' drop (or selection from the file dialog) of multiple files
+   */
+  multiple: PropTypes.bool,
+  /** Maximum file size (in bytes) */
+  maxSize: PropTypes.number,
+  /**
+   * Set accepted file types.
+   * See https://github.com/okonet/attr-accept for more information.
+   */
+  accept: PropTypes.oneOfType([
```
Comment (andresin87) [inline components/atom/upload/src/index.js:150]: format?

**Comment by andresin87** [comment]
add demos with teh new props usage plzz

File: components/atom/upload/src/index.js (hunk 1 above)
Comment (ferransimon) [reply components/atom/upload/src/index.js:150]: accept is the same prop that the wrapped library uses, so for the sake of homogeneity I would keep it as accept.

File: components/atom/upload/src/index.js [hunk 2]
```diff
@@ -22,6 +25,10 @@ const AtomUpload = ({
   status,
   onFilesSelection = () => {},
   textExplanation,
+  actionButton: Button,
```
Comment (andresin87) [inline components/atom/upload/src/index.js:28]: I'm not sure. Where is Button declared?

File: components/atom/upload/src/index.js [hunk 3]
```diff
@@ -30,19 +37,34 @@ const AtomUpload = ({
...
+          <AtomIcon
+            size={ATOM_ICON_SIZES.large}
+            color={ATOM_ICON_COLORS.currentColor}
+            className={classNameIcon}
+          >
+            {IconStatus}
+          </AtomIcon>
+          <div className={CLASS_BLOCK_TEXT}>
+            <h4 className={CLASS_BLOCK_TEXT_MAIN}>{textStatus}</h4>
+            {status === STATUSES.ACTIVE && (textExplanation || Button) && (
```
Comment (AgonisticKatai) [inline components/atom/upload/src/index.js]: Could we put in a more descriptive statement? It's a little confusing to me.

```js
const isActive = status === STATUSES.ACTIVE
const hasTextExplanation = Boolean(textExplanation)
const hasButton = Boolean(Button)

return (
    [...]
    {isActive && (hasTextExplanation || hasButton) && [...]}
)
```

File: components/atom/upload/src/index.js (hunk 2 above)
Comment (ferransimon) [reply components/atom/upload/src/index.js:28]: It is an aliase.

Comment (andresin87) [reply components/atom/upload/src/index.js:28]: i mean, dk what corresponds to 'Button', but it think the default value might be '<Button/>' and not 'Button'

Comment (ferransimon) [reply components/atom/upload/src/index.js:28]: I think I am not getting your point here. It is not a default value it is just a rename of the prop actionButton to be able to use the first letter as a capital letter to detonate it is a component.

File: components/atom/upload/src/index.js (hunk 3 above)
Comment (ferransimon) [reply components/atom/upload/src/index.js]: Done!

File: components/atom/upload/src/index.scss [hunk 4]
```diff
@@ -8,7 +11,7 @@ $fd-atom-upload: column !default;
 $p-atom-upload: $p-l * 3 $p-l * 2 $p-l !default;
 $ai-atom-upload: center !default;
 $jc-atom-upload: center !default;
-$br-atom-upload: 0 !default;
+$br-atom-upload: $bdrs-none !default;
```
Comment (AgonisticKatai) [inline components/atom/upload/src/index.scss]: I think the variable we should use is the `$bdrs-atom-upload`

```suggestion
$br-atom-upload: $bdrs-atom-upload;
```

**Review by AgonisticKatai — APPROVED** [review]

File: components/atom/upload/src/index.scss [hunk 5]
```diff
@@ -8,7 +11,7 @@ $fd-atom-upload: column !default;
 $p-atom-upload: $p-l * 3 $p-l * 2 $p-l !default;
 $ai-atom-upload: center !default;
 $jc-atom-upload: center !default;
-$br-atom-upload: 0 !default;
+$br-atom-upload: $bdrs-atom-upload;
```
Comment (PablitoGS) [inline components/atom/upload/src/index.scss]: You need default! here

**Review by PablitoGS — APPROVED** [review]

**Review by javiauso — APPROVED** [review]

**Review by andresin87 — APPROVED** [review]

**Pull request merged**